src/smartcontracts/interaction.ts

```diff
@@ -141,11 +137,28 @@ export class Interaction {
         return this;
     }
 
-    getEndpointDefinition(): EndpointDefinition {
-        let abi = this.getContract().getAbi();
-        let name = this.getFunction().toString();
-        let endpoint = abi.getEndpoint(name);
-
-        return endpoint;
+    getEndpoint(): EndpointDefinition {
+        return this.getContract().getAbi().getEndpoint(this.getInterpretingFunction());
```

Comment:
When calling a method with a token transfer:
- the _executing_ function is `ESDTTransfer` (or `ESDTNFTTransfer`) - and this is the function that goes into the `function` field of a `Transaction`, which in turn will make the data field contain `"ESDTTransfer@..."`
- the _interpreting_ function is the one we are actually interested in (the actual method we call from the smart contract) - this is the one that's needed most of the time, since we use it to parse the received data (and as such it's used both by `interpretExecutionResults` and `interpretQueryResponse`)

The current usage is correct.